modules/sov-ibc-transfer/src/context.rs

```diff
@@ -72,6 +86,52 @@ impl<'ws, S: Spec> IbcTransferContext<'ws, S> {
             })
     }
 
+    /// Creates a new token with the specified `token_name` and mints an initial
+    /// balance to the `minter_address`.
+    ///
+    /// Note: The mint authority must be held by the `IbcTransfer` module, so the
+    /// `authorized_minters` is set to the `IbcTransfer` address. Also, remember
+    /// that the `token_name` is a denom prefixed with IBC and originates from the
+    /// counterparty chain.
+    fn create_token(
+        &self,
+        token_name: String,
+        minter_address: S::Address,
+    ) -> Result<TokenId, TokenTransferError> {
+        // Using a different salt will result in a different token
+        // address. Since ICS-20 tokens coming from other chains are
+        // guaranteed to have unique names, we don't need to use
+        // different salt values.
+        let salt = 0u64;
+
+        let initial_balance = 0;
+
+        // Make sure to use `ibc_transfer` address as the sender
+        let context = Context::new(
+            self.ibc_transfer.address.clone(),
+            self.sdk_context.sequencer().clone(),
+            self.sdk_context.visible_slot_number(),
+        );
+
+        let new_token_id = self
+            .ibc_transfer
+            .bank
+            .create_token(
+                token_name.clone(),
+                salt,
+                initial_balance,
+                minter_address,
```

Comment:
One last comment. Can we use `self.ibc_transfer.address` here? The `initial_balance` is set to zero anyway.

Reply:
[e9651e1](https://github.com/informalsystems/sovereign-ibc/pull/133/commits/e9651e12c50544acd44a1d4ce4455c65911aebba)